test/integration/request-body/json-with-comment.test.js

```diff
@@ -0,0 +1,169 @@
+            scenario: 'it should remove comments',
+            scenarios: [
+                {
+                    jsoncString: '{/* \nsome comment\n*/\n"a": "value"\n}',
+                    expectedObject: { a: 'value' }
+                },
+                {
+                    jsoncString: '{/*\n"b": "value"\n*/\n"a": "value"\n}',
+                    expectedObject: { a: 'value' }
+                },
+                {
+                    jsoncString: '{/* some */"a": "value"}',
+                    expectedObject: { a: 'value' }
+                },
+                {
+                    jsoncString: '{"a":/* some */ "value"}',
+                    expectedObject: { a: 'value' }
+                },
+                {
+                    jsoncString: '{"a": "value"/* some */}',
+                    expectedObject: { a: 'value' }
+                }
+            ]
+        },
+        {
+            suitName: 'when quotes with escape character present',
+            scenario: 'it should handle it properly',
+            /* eslint-disable no-useless-escape */
+            scenarios: [
+                {
+                    jsoncString: '{//test\n"a": "val\\\"ue"}',
+                    expectedObject: { a: 'val"ue' }
+                },
+                {
+                    jsoncString: '{//test\n"a\\\"a": "val\\\"ue"}',
+                    expectedObject: { 'a"a': 'val"ue' }
+                }
+            ]
+        },
+        {
+            suitName: 'when the json is invalid',
+            scenario: 'it should handle it properly',
+            scenarios: [
+                {
+                    jsoncString: '{//test\n"a": "value",}',
+                    expectedObject: '{\n"a": "value",}'
+                }
+            ]
+        },
+        {
+            suitName: 'when content-type header is extension of application/json',
+            scenario: 'it should remove comments',
+            scenarios: [
+                {
+                    contentTypeHeader: 'application/json-merge-patch',
+                    jsoncString: '{//test\n"a": "value",}',
+                    expectedObject: '{\n"a": "value",}'
+                }
+            ]
+        },
+        {
+            suitName: 'when content-type header is not application/json',
+            scenario: 'it should not remove the comments',
+            scenarios: [
+                {
+                    contentTypeHeader: 'text/plain',
+                    jsoncString: '{//test\n"a": "value",}',
+                    expectedObject: '{//test\n"a": "value",}'
+                }
+            ]
+        }
+    ];
+
+
+testSuit.forEach((test) => {
+    describe(test.suitName, function () {
+        let testrun, URL_HEADER;
+
+        before(function () {
+            URL_HEADER = global.servers.http + '/echo/post';
+        });
+
+        test.scenarios.forEach((scenario) => {
+            describe(test.scenario, function () {
+                before(function (done) {
```

Comment:
` describe(test.scenario, function () {` will this print an object for the describe string ?

Reply:
it's `test.scenario` not scenario. So I thought to have a default scenario name for test suit. Later can add specific to any scenario like `describe(scenario.name || test.scenario`

`
      {
            suitName: 'when single line comment present',
            scenario: 'it should remove comments',
            scenarios: [
                {
                    jsoncString: '{// some comment\n"a": "value"\n}',
                    expectedObject: { a: 'value' }
                },
`